packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3315,10 +3315,13 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
   }
 
   /// Toggles the visibility of the toolbar.
-  void toggleToolbar() {
-    assert(_selectionOverlay != null);
+  void toggleToolbar([bool hideHandles = true]) {
+    if (_selectionOverlay == null) {
+      _createSelectionOverlay();
+    }
```

Comment:
`final selectionOverlay = _selectionOverlay ??= createSelectionOverlay()` to avoid the force cast?